cluster/v1alpha1/types_rolloutstrategy.go

```diff
@@ -0,0 +1,83 @@
+type RolloutStrategy struct {
+	// Rollout strategy Types are All, Progressive and ProgressivePerGroup
+	// 1) All means apply the workload to all clusters in the decision groups at once.
+	// 2) Progressive means apply the workload to the selected clusters progressively per cluster. The workload will not be applied to the next cluster unless one of the current applied clusters reach the successful state or timeout.
+	// 3) ProgressivePerGroup means apply the workload to decisionGroup clusters progressively per group. The workload will not be applied to the next decisionGroup unless all clusters in the current group reach the successful state or timeout.
+	// +kubebuilder:validation:Enum=All;Progressive;ProgressivePerGroup
+	// +kubebuilder:default:=All
+	// +optional
+	Type string `json:"type,omitempty"`
+
+	// All define required fields for RolloutStrategy type All
+	// +optional
+	All *RolloutAll `json:"all,omitempty"`
+
+	// Progressive define required fields for RolloutStrategy type Progressive
+	// +optional
+	Progressive *RolloutProgressive `json:"progressive,omitempty"`
+
+	// ProgressivePerGroup define required fields for RolloutStrategy type ProgressivePerGroup
+	// +optional
+	ProgressivePerGroup *RolloutProgressivePerGroup `json:"progressivePerGroup,omitempty"`
+}
+
+// RolloutAll is a RolloutStrategy Type
+type RolloutAll struct {
+	// +optional
+	Timeout Timeout `json:",inline"`
+}
+
+// RolloutProgressivePerGroup is a RolloutStrategy Type
+type RolloutProgressivePerGroup struct {
+	// List of the decision groups names or indexes to apply the workload first and fail if workload did not reach successful ((not proceed to apply workload to other decision groups/clusters).
+	// +optional
+	MandatoryDecisionGroups []MandatoryDecisionGroup `json:"mandatoryDecisionGroups,omitempty"`
+
+	// +optional
+	Timeout Timeout `json:",inline"`
+}
+
+// RolloutProgressive is a RolloutStrategy Type
+type RolloutProgressive struct {
+	// List of the decision groups names or indexes to apply the workload first and fail if workload did not reach successful ((not proceed to apply workload to other decision groups/clusters).
+	// +optional
+	MandatoryDecisionGroups []MandatoryDecisionGroup `json:"mandatoryDecisionGroups,omitempty"`
+
+	// MaxConcurrency is the max number of clusters to deploy workload concurrently. The default value for MaxConcurrency is determined from the clustersPerDecisionGroup defined in the placement->DecisionStrategy.
+	// +kubebuilder:validation:Pattern="^((100|[0-9]{1,2})%|[0-9]+)$"
+	// +optional
+	MaxConcurrency string `json:"maxConcurrency,omitempty"`
+
+	// +optional
+	Timeout Timeout `json:",inline"`
+}
+
+// Timeout to consider while applying the workload.
+type Timeout struct {
+	// Timeout define how long workload applier controller will wait till workload reach successful state in the cluster. Only considered for Rollout Type Progressive and ProgressivePerGroup.
```

Comment:
This comment is inconsistent with the above type `RolloutAll` which has timeout field.
```
// RolloutAll is a RolloutStrategy Type
type RolloutAll struct {
	// +optional
	Timeout Timeout `json:",inline"`
}
```